### PR TITLE
Update the is_causal explaination in the SDPA doc

### LINKED ISSUE #126873: [docs] scaled_dot_product_attention is_causal description is misleading
### 📚 The doc issue

In scaled_dot_product_attention parameter description, the current wording 
"[is_causal](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) (bool) – If true, assumes **upper left causal attention masking** and errors if both attn_mask and is_causal are set."
is misleading. I guess there is a mistake?

### Suggest a potential alternative/fix

It should be lower/left masking (or upper/right depending on the wording)

cc @svekars @brycebortree @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki

### PR DESCRIPTION
**Pull request by lancerts**
Fixes #126873

cc @svekars @brycebortree @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### CODE REVIEW

**Review by drisspg — APPROVED** [review]
Thank you!

@pytorchbot merge
